src/lib/modules/console/index.ts

```diff
@@ -0,0 +1,226 @@
+          this.logger.warn("[DEPRECATED] In future versions of embark, we expect the console command to return an object " +
+            "having 2 functions: match and process." +
+            " The documentation with example can be found here: https://embark.status.im/docs/plugin_reference.html#embark-registerConsoleCommand-callback-options");
+          return callback(null, pluginResult);
+        }
+      } else if (pluginResult.match()) {
+        return pluginResult.process(callback);
+      }
+    }
+
+    const output = this.processEmbarkCmd(cmd);
+    if (output) {
+      return callback(null, output);
+    }
+
+    try {
+      this.events.request("runcode:eval", cmd, callback);
+    } catch (e) {
+      if (this.ipc.connected && this.ipc.isClient()) {
+        return this.ipc.request("console:executeCmd", cmd, callback);
+      }
+      callback(e);
+    }
+  }
+
+  private registerEmbarkJs() {
+    this.events.emit("runcode:register", "IpfsApi", IpfsApi, false);
+    this.events.emit("runcode:register", "Web3", Web3, false);
+    this.events.emit("runcode:register", "EmbarkJS", EmbarkJS, false);
+
+    EmbarkJS.Blockchain.done = true;
+    if (this.ipc.connected) {
+      return;
+    }
+
+    this.events.once("code-generator-ready", () => {
+      this.events.request("code-generator:embarkjs:provider-code", (code: string) => {
+        const func = () => {};
+        this.events.request("runcode:eval", code, func, true);
+        this.events.request("runcode:eval", this.getInitProviderCode(), func, true);
+      });
+    });
+  }
+
+  private getInitProviderCode() {
+    const codeTypes: any = {
+      communication: this.config.communicationConfig || {},
+      names: this.config.namesystemConfig || {},
+      storage: this.config.storageConfig || {},
+    };
+
+    return this.plugins.getPluginsFor("initConsoleCode").reduce((acc: any, plugin: any) => {
+      Object.keys(codeTypes).forEach((codeTypeName: string) => {
+        (plugin.embarkjs_init_console_code[codeTypeName] || []).forEach((initCode: any) => {
+          const [block, shouldInit] = initCode;
+          if (shouldInit.call(plugin, codeTypes[codeTypeName])) {
+            acc += block;
+          }
+        });
+      });
+      return acc;
+    }, "");
+  }
+
+  private registerConsoleCommands() {
+    this.embark.registerConsoleCommand((cmd: string, options?: any) => {
+      const [cmdName, length] = cmd.split(" ");
+      return {
+        match: () => cmdName === "history",
+        process: (callback: any) => this.getHistory(length, callback),
+      };
+    });
+  }
+
+  private loadHistory() {
+    if (fs.existsSync(this.cmdHistoryFile)) {
+      fs.readFileSync(this.cmdHistoryFile)
+        .toString()
+        .split("\n")
+        .reverse()
+        .forEach((cmd: string) => { this.history.push(cmd); });
+    }
+  }
+
+  private getHistory(historySize: any, callback: any) {
```

Comment:
For `historySize` is there a way to say "either a string or an int"? Also, same for `callback`.

Reply:
Yes, that'd be done with `historySize: string | number`. This could also be a type alias: `type HistorySize = string | number;`